tests/assertions/is_substring_of.zsh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env zunit
+
```

Comment:
I agree, there are quite a few complications involved with empty strings so it would be best it it fails and the tests explicitly test an empty string instead